accepted/0042-isolated-mode.md

```diff
@@ -0,0 +1,347 @@
+   │      │
+   │      └───> B @ 1.0.0
+   │
+   └───> bar (workspace)
+          ┬
+          │
+          ├───> A @ 1.0.0
+          │       ┬
+          │       │
+          │       └───> B @ * (peer dependency)
+          │
+          └───> B @ 2.0.0
+
+
+```
+
+#### Installation on disk
+
+```
+  root /
+   ┬
+   │
+   ├─> node_modules / .npm / ─┬─> A@1.0.0+B@1.0.0-21f95f7 / node_modules / ─┬─> A [ content of package A ]
+   │                          │                                             │
+   │                          │                                             └─> B ( symlink to ../../B@1.0.0-0d98ab/node_modules/B )
+   │                          │
+   │                          ├─> A@1.0.0+B@2.0.0-66fe689 / node_modules / ─┬─> A  [ content of package A ]
+   │                          │                                             │
+   │                          │                                             └─> B ( symlink to ../../B@2.0.0-a2ea56/node_modules/B )
+   │                          │
+   │                          ├─> B@1.0.0-0d98ab / node_modules / B / ───> [ content of package B (v1) ]
+   │                          │
+   │                          └─> B@2.0.0-a2ea56 / node_modules / B / ───> [ content of package B (v2) ]
+   │
+   └─> workspaces / ─┬─> foo / ─┬─> [ content of workspace foo ]
+                     │          │
+                     │          └───> node_modules / ─┬─> A ( symlink to ../../../node_modules/.npm/A@1.0.0+B@1.0.0-21f95f7/node_modules/A )
+                     │                                │
+                     │                                └─> B ( synlink to ../../../node_modules/.npm/B@1.0.0-0d98ab/node_modules/B )
+                     │
+                     └─> bar / ─┬─> [ content of workspace bar ]
+                                │
+                                └───> node_modules / ─┬─> A ( symlink to ../../../node_modules/.npm/A@1.0.0+B@2.0.0-66fe689/node_modules/A )
+                                                      │
+                                                      └─> B ( synlink to ../../../node_modules/.npm/B@2.0.0-a2ia56/node_modules/B )
+
+```
+
+## Performance
+
+Compared to the current npm installation strategy, this proposal reduces package duplication, making the installation process faster. On a prototype, this installation strategy brought down the install time from 6 to 2 minutes on a large monorepo of 500+ workspaces.
+
+## Configuration
+
+The `mode` can be enabled by a CLI option when running `npm install` or by a setting in `.npmrc`.
+
+## Compatibility
+
+### Different versions of npm
+
+The `isolated-mode` will use the same lockfile as `hoisted` mode. Developers can use either mode without changing the lockfile. This makes `isolated-mode` backward compatible and easy to implement. The implementation of `isolated-mode` will reside in the reification, it will take the ideal-tree and apply it on disk using symlink instead of hoisting.
```

Comment:
What happens if i use one reification, and then run npm install with another? Will it be able to trivially go back and forth without incident?

Reply:
it should be able to do the right thing, meaning change from hoisted to isolated and back with no leftover from the previous installation.